docs/intro/from-ground-up.rst

```diff
@@ -0,0 +1,542 @@
+.. _`from-ground-up`:
+
+===========================
+web-poet from the ground up
+===========================
+
+This tutorial explains the motivation behind web-poet, and its main
+concepts. You would learn
+
+* what are the issues which ``web-poet`` addresses, why does the library exist;
+* how ``web-poet`` addresses these issues;
+* what is a Page Object, and what is a page object input;
+* how are libraries like `scrapy-poet`_ implemented, and what are they for.
+
+Reusable web scraping code
+==========================
+
+Forget about web-poet for a minute. Let's say you're writing code to scrape
+a book web page from `books.toscrape.com <http://books.toscrape.com/>`_:
+
+.. code-block:: python
+
+    import requests
+    import parsel
+
+
+    def extract_book(url):
+        """ Extract book information from a book page on
```

Comment:
Should we follow our style pattern and get rid of those leading white spaces on examples’ docstrings as well?

Reply:
argh. Following one style would be better. I prefer to have a whitspace, and/or a blank line in multi-line docstrings, because otherwise it looks too condensed to me. 

PEP257 says blank line is fine, while it seems an extra whitespace is not. So a proposal: avoid extra whitespace in one-line docstrings; use a blank empty line in multi-line docstrings.

Reply:
I also wonder if we should be religious about it; such conventions can be a barrier to contribution (we'll be pointing future contributors to these issues), and they may be hard to check automatically (unlike e.g. line length limit).

Reply:
Sounds good to me.